### PR TITLE
preserve chunked data when creating DataArray from DataArray

### LINKED ISSUE #5983: preserve chunked data when creating DataArray from itself
**What happened**:

When creating a new DataArray from a DataArray with chunked data, the underlying dask array is converted to a numpy array.  

**What you expected to happen**:

I expected the underlying dask array to be preseved when creating a new DataArray instance.

**Minimal Complete Verifiable Example**:

```python
import xarray as xr
import numpy as np
from dask import array

d = np.ones((10, 10))
x = array.from_array(d, chunks=5)

da = xr.DataArray(x) # this is chunked
xr.DataArray(da) # this is not chunked anymore
```

**Anything else we need to know?**:

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>
INSTALLED VERSIONS
------------------
commit: None
python: 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0]
python-bits: 64
OS: Linux
OS-release: 5.11.0-40-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: ('en_US', 'UTF-8')
libhdf5: 1.10.6
libnetcdf: 4.7.4

xarray: 0.19.0
pandas: 1.3.3
numpy: 1.20.3
scipy: 1.7.1
netCDF4: 1.5.6
pydap: None
h5netcdf: 0.11.0
h5py: 3.2.1
Nio: None
zarr: 2.10.1
cftime: 1.5.0
nc_time_axis: None
PseudoNetCDF: None
rasterio: 1.2.6
cfgrib: None
iris: None
bottleneck: 1.3.2
dask: 2021.09.1
distributed: 2021.09.1
matplotlib: 3.4.3
cartopy: 0.19.0.post1
seaborn: 0.11.2
numbagg: None
pint: None
setuptools: 58.0.4
pip: 21.2.4
conda: 4.10.3
pytest: 6.2.5
IPython: 7.27.0
sphinx: 4.2.0

</details>

### PR DESCRIPTION
**Pull request by FabianHofmann**
- [x] Closes #5983 
- [x] Tests added
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: xarray/tests/test_dataarray.py [hunk 1]
```diff
@@ -410,6 +410,21 @@ def test_constructor_from_self_described(self):
...
+        import dask.array as da
+
+        data = da.from_array([[-0.1, 21], [0, 2]])
+        expected = DataArray(
+            data,
+            coords={"x": ["a", "b"], "y": [-1, -2]},
+            dims=["x", "y"],
+            name="foobar",
+            attrs={"bar": 2},
+        )
```
Comment (Illviljan) [inline xarray/tests/test_dataarray.py:424]:
```suggestion
        expected = DataArray(
            [[-0.1, 21], [0, 2]],
            coords={"x": ["a", "b"], "y": [-1, -2]},
            dims=["x", "y"],
            name="foobar",
            attrs={"bar": 2},
        ).chunk()
```
I think this should be the same.

Comment (FabianHofmann) [reply xarray/tests/test_dataarray.py:424]: Indeed

File: xarray/tests/test_dataarray.py [hunk 2]
```diff
@@ -410,6 +410,18 @@ def test_constructor_from_self_described(self):
...
+    def test_constructor_from_self_described_chunked(self):
+        expected = DataArray(
+            [[-0.1, 21], [0, 2]],
+            coords={"x": ["a", "b"], "y": [-1, -2]},
+            dims=["x", "y"],
+            name="foobar",
+            attrs={"bar": 2},
+        ).chunk()
+        actual = DataArray(expected)
+        assert_identical(expected, actual)
```
Comment (mathause) [inline xarray/tests/test_dataarray.py:423]: This does not check if the chunks are equal.
```suggestion
        assert_chunks_equal(expected, actual)
```

Comment (FabianHofmann) [reply xarray/tests/test_dataarray.py:423]: True, but shouldn't we check both, that the array is identical and the chunks are the same? If I see correctly, `assert_chunks_equal` does not compare assigned attributes, like `coords`, `attrs`.

Comment (mathause) [reply xarray/tests/test_dataarray.py:423]: Yes checking both is good.

Comment (FabianHofmann) [reply xarray/tests/test_dataarray.py:424]: alright, both tests are there now

**Review by mathause — APPROVED** [review]
Looks good to me. I can merge in a few days if there are no objections.

You could also add a short note to [whats-new](https://github.com/pydata/xarray/blob/main/doc/whats-new.rst) - probably under bug fixes.

**Comment by mathause** [comment]
@dcherian you had an opinion on this in #5983 - would you rather raise a `ValueError` here and advise to pass `da.data`?

**Comment by dcherian** [comment]
How about we raise DeprecationWarning for now asking the user to pass in `.data` instead.

**Comment by mathause** [comment]
> How about we raise DeprecationWarning for now asking the user to pass in `.data` instead.

Makes sense - how about:

```python
msg = (
    "Passing a `DataArray` object to the `DataArray` constructor is ambiguous.\n"
    "Pass `da.data`, or use `da.copy(deep=True)`, or one of the `xarray.*_like` (e.g. "
    "`xarray.zeros_like`) functions instead,\n"
    "depending on your use case."
)
warnings.warn(msg, FutureWarning)
```

**Comment by max-sixty** [comment]
> How about we raise DeprecationWarning for now asking the user to pass in `.data` instead.

Does this mean `xr.DataArray(da)` would be deprecated? That seems like a very common case. Sorry for joining late but could I ask why this is necessary, if my assumption is right? I had been thinking that is broadly equivalent to a copy.

**Comment by keewis** [comment]
for reference, the situation we already deprecated and removed was passing `(["x", "y"], da)` to `as_variable` (for example, when passed to the `coords` parameter of `DataArray`), which would create a new variable with new dimension names, but drop the old attrs and the name / coords *and* it would convert `var` to `numpy`, which could trigger a computation if the data was a `dask` array (or raise / emit a warning for other array types).

In contrast to that, I think we *can* keep `xr.DataArray(da)` working, but *only* if it's the only argument (i.e. I don't think we should allow `xr.DataArray(da, dims=("x", "y"))` or something similar – those might still be useful, but I'd make them separate methods instead of overloading the constructor).

However, this might make the constructor a bit more difficult to understand so if we don't think that's worth the (minor) inconvenience:
```python
da if isinstance(da, xr.DataArray) else xr.DataArray(da)
```
we can go ahead with the deprecation.

**Comment by max-sixty** [comment]
> In contrast to that, I think we _can_ keep `xr.DataArray(da)` working, but _only_ if it's the only argument (i.e. I don't think we should allow `xr.DataArray(da, dims=("x", "y"))` or something similar – those might still be useful, but I'd make them separate methods instead of overloading the constructor).

Perfect, very much agree, thank you for the clear description @keewis !

**Comment by mathause** [comment]
Any objections to merging this as is? If we don't disallow `xr.DataArray(da)` this is a clear advantage over the current version.

However, 

```python
xr.DataArray(da).data is da.data # True
xr.DataArray(da.data).data is da.data # True
# while
da.copy().data is da.data # False (deep=True is the default)
```
might just be surprising in some situations.


(Just for completeness - `xr.DataArray(da, **kwargs)` does a good job at erroring if the `kwargs` are not compatible with `da`.)

**Comment by dcherian** [comment]
> (Just for completeness - xr.DataArray(da, **kwargs) does a good job at erroring if the kwargs are not compatible with da.)

Sounds good to me then!

**Comment by Illviljan** [comment]
Thank you, @FabianHofmann !

**Comment by FabianHofmann** [comment]
It was a pleasure!

**Pull request merged**